packages/help-center/src/components/help-center-contact-form.tsx

```diff
@@ -108,52 +107,109 @@ function openPopup( event: React.MouseEvent< HTMLButtonElement > ): Window {
 const ContactForm: React.FC< ContactFormProps > = ( { mode, onBackClick } ) => {
 	const [ openChat, setOpenChat ] = useState( false );
 	const [ contactSuccess, setContactSuccess ] = useState( false );
+	const [ forumTopicUrl, setForumTopicUrl ] = useState( '' );
+	const [ hideSiteInfo, setHideSiteInfo ] = useState( false );
 	const locale = useLocale();
-	const { isLoading: submittingTicket, mutateAsync } = useSubmitTicketMutation();
-	const { siteId, subject, message, otherSiteURL } = useSelect( ( select ) => {
+	const { isLoading: submittingTicket, mutateAsync: submitTicket } = useSubmitTicketMutation();
+	const { isLoading: submittingTopic, mutateAsync: submitTopic } = useSubmitForumsMutation();
+	const [ sitePickerChoice, setSitePickerChoice ] = useState< 'CURRENT_SITE' | 'OTHER_SITE' >(
+		'CURRENT_SITE'
+	);
+
+	const { selectedSite, subject, message, userDeclaredSiteUrl } = useSelect( ( select ) => {
 		return {
-			siteId: select( STORE_KEY ).getSiteId(),
+			selectedSite: select( STORE_KEY ).getSite(),
 			subject: select( STORE_KEY ).getSubject(),
 			message: select( STORE_KEY ).getMessage(),
-			otherSiteURL: select( STORE_KEY ).getOtherSiteURL(),
+			userDeclaredSiteUrl: select( STORE_KEY ).getUserDeclaredSiteUrl(),
 		};
 	} );
-	const { hasCookies, isLoading: loadingCookies } = useHas3PC();
 
-	const isLoading = loadingCookies || submittingTicket;
+	const { setSite, setUserDeclaredSiteUrl, setUserDeclaredSite, setSubject, setMessage, setPopup } =
+		useDispatch( STORE_KEY );
+
+	const {
+		result: ownershipResult,
+		isLoading: isAnalysisLoading,
+		site: userDeclaredSite,
+	} = useSiteAnalysis( userDeclaredSiteUrl );
 
-	const { setSiteId, setOtherSiteURL, setSubject, setMessage, setPopup } = useDispatch( STORE_KEY );
+	// record the resolved site
+	useEffect( () => {
+		if ( userDeclaredSite ) {
+			setUserDeclaredSite( userDeclaredSite );
+		}
+	}, [ userDeclaredSite, setUserDeclaredSite ] );
+
+	const { hasCookies, isLoading: loadingCookies } = useHas3PC();
+
+	const isLoading = loadingCookies || submittingTicket || submittingTopic;
 
 	const formTitles = titles[ mode ];
 
+	const currentSite = useSelect( ( select ) =>
+		select( SITE_STORE ).getSite( window._currentSiteId )
+	);
+
+	let supportSite: typeof currentSite;
+
+	// if the user picked "other site", force them to declare a site
+	if ( sitePickerChoice === 'OTHER_SITE' ) {
+		supportSite = userDeclaredSite;
+	} else {
+		supportSite = selectedSite || currentSite;
+	}
+
 	function handleCTA( event: React.MouseEvent< HTMLButtonElement > ) {
-		switch ( mode ) {
-			case 'CHAT': {
-				if ( hasCookies ) {
-					setOpenChat( true );
+		if ( supportSite ) {
+			switch ( mode ) {
+				case 'CHAT': {
+					if ( hasCookies ) {
+						setOpenChat( true );
+						break;
+					} else {
+						const popup = openPopup( event );
+						setPopup( popup );
+					}
+					break;
+				}
+				case 'EMAIL': {
+					const ticketMeta = [
+						'How can you help: ' + message,
```

Comment:
This is sent by the user? Should it be `How we can help you:` ?

Reply:
Yes, but then forward the message to the HE like this, and this is the message we show the HE. Confused me too ([this is not my decision](https://github.com/Automattic/wp-calypso/blob/trunk/client/me/help/help-contact/index.jsx#L182)).